sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetastoreCatalogSuite.scala

```diff
@@ -44,9 +44,16 @@ class HiveMetastoreCatalogSuite extends TestHiveSingleton with SQLTestUtils {
   }
 
   test("duplicated metastore relations") {
-    val df = spark.sql("SELECT * FROM src")
-    logInfo(df.queryExecution.toString)
-    df.as('a).join(df.as('b), $"a.key" === $"b.key")
+    val originalCreateHiveTable = TestHive.conf.createHiveTableByDefaultEnabled
+    try {
+      TestHive.conf.setConf(SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED, true)
+      val df = spark.sql("SELECT * FROM src")
```

Comment:
where do we create `src`?

Reply:
It's created in `TestHive`. `TestHive` should always create hive tables, can we update it to write `STORED AS`?

Reply:
I believe many changes can be reverted if we fix `TestHive`.

Reply:
Instead of modifying the query, how about setting legacy mode for the `TestHiveContext` ?
https://github.com/apache/spark/blob/abba53e78b5556016a9d6935cfc65a9299a78977/sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHive.scala#L52-L73

Those queries are migrated from hive directly, so, I think it might be better not to change it ?

> I believe many changes can be reverted if we fix TestHive.

Yeah, I think so. At the beginning I'm not sure whether it has side affects for other tests if we modify the query directly, so I fixed them one by one. If we could fix them in a centralize place, it would be better.

Reply:
> how about setting legacy mode for the TestHiveContext

Then all the tests in hive module do not apply the new change, which decrease the test coverage.

I don't see them as "query". They are just data preparations and we should create hive tables.